### ISSUE TITLE
As a repository operator I want to validate maDMPs against a schema to ensure their compliance.

### ISSUE THREAD

**Issue by TomMiksa** [samihabakri/D2S-Dataverse#16]
JSON Schema for the RDA DMP Common Standard
https://json-schema.org
https://github.com/RDA-DMP-Common/RDA-DMP-Common-Standard

**Comment by AhmadHirthani** [comment]
https://ajv.js.org/
https://github.com/epoberezkin/ajv

**Comment by TomMiksa** [comment]
You can use this one to play around: https://www.jsonschemavalidator.net
First step is creating a JSON schema out of this: https://github.com/RDA-DMP-Common/RDA-DMP-Common-Standard/blob/master/docs/index.md

**Comment by AhmadHirthani** [comment]
I validated maDMP against the RDA DMP Common Standard schema.
Used tools:
AJV script as a validating tool
PHP scripting language to implement back-end coding

**Comment by TomMiksa** [comment]
Can you provide a link to the schema? Please put it on GitHub and do not provide about a simple readme. 

And please also include the maDMP instances that you have used for testing.

**Comment by AhmadHirthani** [comment]
I uploaded the schema and the testing msDMP.
Schema file link: https://github.com/samihabakri/D2S-Dataverse/blob/master/OtherServices/validate_maDMP/schema.json

Testing maDMP file link: https://github.com/samihabakri/D2S-Dataverse/blob/master/OtherServices/validate_maDMP/testing%20msDMP

I use this tool https://www.jsonschema.net/ to convert this maDMP https://github.com/RDA-DMP-Common/RDA-DMP-Common-Standard/blob/master/examples/JSON/ex8-dmp-minimal-content.json into a schema.

**Comment by TomMiksa** [comment]
The goal of having a model documentation converted into schema is to be able to validate any DMP - not just the minimal. For this reason, the schema has to include all fields.

**Comment by AhmadHirthani** [comment]
Examples in this link https://github.com/RDA-DMP-Common/RDA-DMP-Common-Standard/tree/master/examples/JSON have a problem with the field names, specifically the word separators. 
I have got fixed examples for testing.

**Comment by TomMiksa** [comment]
Yes, I know. We will have to replace the examples. However, I dont see in your schema anythign about Metadata, Distribution, License, etc. - it's incomplete.

**Comment by AhmadHirthani** [comment]
I create this schema for testing using automatic tool https://jsonschema.net/. Now I am working on creating my own schema.

**Comment by AhmadHirthani** [comment]
I finished creating maDMP schema. I uploaded it. I hope it will be a complete maDMP schema.
Tool is now running.

**Comment by AhmadHirthani** [comment]
There is a problem in validating "date-time" type. Since I must make some changes in the dates of  maDMP examples to this type "2018-11-13T20:20:39+00:00" https://json-schema.org/understanding-json-schema/reference/string.html . There is an issue already opened on the RDA DMP common standards project https://github.com/RDA-DMP-Common/RDA-DMP-Common-Standard/issues/6. I suggest changing the "date-time" format in examples to ISO 8601 format.

**Comment by AhmadHirthani** [comment]
To show errors in validating process I must use the function require('ajv-errors') which is not possible in Javascript. So I used http://browserify.org/ which is a tool that enables me to use require() through NodeJS. Using browserify generates the file bundle.js. I also uploaded a Full maDMP testing file.
